### PR TITLE
Feature/cof 17 set up e2e tests with cypress

### PR DESCRIPTION
**Pull request by AP2610**
Added Cypress to the project.

### CODE REVIEW

**Review by LuukRos — APPROVED** [review]

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// declare global {
+//   namespace Cypress {
+//     interface Chainable {
+//       login(email: string, password: string): Chainable<void>
+//       drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
+//       dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
+//       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
+//     }
+//   }
+// }
```
Comment (NeeLonGithub) [inline cypress/support/commands.ts]:
```suggestion
// }

```

File: cypress/support/component-index.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width,initial-scale=1.0">
+    <title>Components App</title>
+    <!-- Used by Next.js to inject CSS. -->
+    <div id="__next_css__DO_NOT_USE__"></div>
+  </head>
+  <body>
+    <div data-cy-root></div>
+  </body>
+</html>
```
Comment (NeeLonGithub) [inline cypress/support/component-index.html]:
```suggestion
</html>

```

File: cypress/support/component.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    interface Chainable {
+      mount: typeof mount
+    }
+  }
+}
+
+Cypress.Commands.add('mount', mount)
+
+// Example use:
+// cy.mount(<MyComponent />)
```
Comment (NeeLonGithub) [inline cypress/support/component.ts]:
```suggestion
// cy.mount(<MyComponent />)

```

File: cypress/support/e2e.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// Import commands.js using ES2015 syntax:
+import './commands'
+
+// Alternatively you can use CommonJS syntax:
+// require('./commands')
```
Comment (NeeLonGithub) [inline cypress/support/e2e.ts]:
```suggestion
// require('./commands')

```

File: src/components/button/Button.tsx [hunk 5]
```diff
@@ -5,7 +5,7 @@ interface ButtonProps {
   /**
    * Is this the principal call to action on the page?
    */
-  primary?: boolean;
+  isPrimary?: boolean;
```
Comment (NeeLonGithub) [inline src/components/button/Button.tsx]: Out of scope so non-blocking, but I'd prefer if we don't use booleans for things that just happens to have only two options at the moment.

```suggestion
  type?: 'primary' | 'secondary';
```

File: src/components/button/Button.tsx [hunk 6]
```diff
@@ -27,8 +27,8 @@ interface ButtonProps {
 /**
  * Primary UI component for user interaction
  */
-export const Button = ({ primary = false, size = 'medium', backgroundColor, label, ...props }: ButtonProps) => {
-  const mode = primary ? 'storybook-button--primary' : 'storybook-button--secondary';
+export const Button = ({ isPrimary = false, size = 'medium', backgroundColor, label, ...props }: ButtonProps) => {
+  const mode = isPrimary ? 'storybook-button--primary' : 'storybook-button--secondary';
```
Comment (NeeLonGithub) [inline src/components/button/Button.tsx]: And then here:
```suggestion
export const Button = ({ type = 'secondary', size = 'medium', backgroundColor, label, ...props }: ButtonProps) => {
  const mode = type === 'primary' ? 'storybook-button--primary' : 'storybook-button--secondary';
```

**Review by NeeLonGithub — APPROVED** [review]
Few 'suggestions' 😊

**Review by ambertaal — APPROVED** [review]
Looks good to me!

**Pull request merged**